src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double side;
+
+    public Square(String color, double side) {
```

Comment:
Instead of generating the 'side' value inside this class, consider passing the value that is already generated by the `FigureSupplier`. This will make the code more flexible and adhere to the single responsibility principle.